state-chain/pallets/cf-elections/src/benchmarking.rs

```diff
@@ -60,6 +57,30 @@ mod benchmarks {
 		);
 	}
 
+	#[benchmark]
+	fn provide_shared_data() {
+		let validator_id = ready_validator_for_vote::<T, I>();
+
+		let elections = Pallet::<T, I>::electoral_data(&validator_id).unwrap().current_elections;
+		let next_election = elections.into_iter().next().unwrap();
+
+		assert_ok!(Pallet::<T, I>::vote(
+			RawOrigin::Signed(validator_id.clone().into()).into(),
+			BoundedBTreeMap::try_from(
+				[(next_election.0, T::ElectoralSystem::benchmark_authority_vote())]
+					.into_iter()
+					.collect::<BTreeMap<_, _>>(),
+			)
+			.unwrap(),
+		));
+
+		#[extrinsic_call]
+		provide_shared_data(
+			RawOrigin::Signed(validator_id.into()),
+			T::ElectoralSystem::benchmark_shared_data(),
+		);
```

Comment:
Something like this to see that we actually inserted the shared data, which is the most expensive path.
```
		assert_eq!(
			SharedData::<T, I>::get(&SharedDataHash::of(&T::ElectoralSystem::benchmark_shared_data())),
			Some(T::ElectoralSystem::benchmark_shared_data())
		);
```

Reply:
I just pushed it since I tried it - and made the diagram text which I think was removed when the base branch was rebased